### ISSUE TITLE
Type predicate

### ISSUE THREAD

**Issue by andy-ms** [JamesHenry/tsep-babylon-test#15]
### Code

```ts
declare function f(x: any): x is boolean;
```

### Eslint

```json
"type": "TSFirstTypeNode",
```

### Babylon

```json
"type": "TSTypePredicate",
```

**Comment by JamesHenry** [comment]
Agreed, thanks!

**Comment by JamesHenry** [comment]
I have submitted a fix for this: https://github.com/eslint/typescript-eslint-parser/pull/283, however it requires a small update to babylon too to match the fix for typeAnnotation structure in #1

**Comment by JamesHenry** [comment]
Looks like this change was made all round 😄
